lib/spack/spack/cmd/view.py

```diff
@@ -158,11 +164,22 @@ def view(parser, args):
     specs = spack.cmd.parse_specs(args.specs)
     path = args.path[0]
 
+    if args.projection_file:
+        if os.path.exists(args.projection_file):
+            with open(args.projection_file, 'r') as f:
+                projections_data = s_yaml.load(f)
+                validate(projections_data, spack.schema.projections.schema)
+                ordered_projections = projections_data['projections']
+        else:
+            tty.die('Specified projection file does not exist.')
```

Comment:
I agree with https://github.com/spack/spack/pull/9679#discussion_r245464233 that the program should stop if the file doesn't exist; my primary concern with this is that if we do error handling in commands when they could be done at a higher level, we will end up with repeat code that complicates the logic. I think that for command line arguments that specify a file, we can make a reusable verifier that is specified when defining the argument with argparse.

Anyhow feel free to leave this as is (I think this is minor), but I got the impression I wasn't clear about my complaint so I wanted to clarify it.

Reply:
Ok I understand the complaint better now.

I think using an argparse type to check for this is a good idea, but not to check for the schema (as most commands don't use schema validation. I've implemented that.